eos/utils/parameters.cc

```diff
@@ -640,9 +691,17 @@ namespace eos
         return _imp->parameters[id];
     }
 
-    Parameter
+    void
     Parameters::declare(const QualifiedName & name, const std::string & latex, Unit unit, const double & value, const double & min, const double & max)
     {
+        ParameterDefaults::instance()->insert(name, Parameter::Template { name, min, value, max, latex, unit });
+    }
+
+    Parameter
+    Parameters::declare_and_insert(const QualifiedName & name, const std::string & latex, Unit unit, const double & value, const double & min, const double & max)
+    {
+        ParameterDefaults::instance()->insert(name, Parameter::Template { name, min, value, max, latex, unit });
+
```

Comment:
Shouldn't this appear after line 710, after the comment `// create new parameter`?

Reply:
Well, this declares it but does not insert it yet. I will adapt the comments, which were taken without modification from the previous code.

Reply:
But the point was that the parameter might already have been declared.

Reply:
Ahh. Good point. Something is clearly going awry when we use this method to ``declare_and_insert`` an already available parameter, contrary to the pure ``declare`` which might simply override a different analysis file's parameter declarations. I will deal with it.